roles/ceph-common/templates/ceph.conf.j2

```diff
@@ -8,59 +8,62 @@ auth service required = none
 auth client required = none
 auth supported = none
 {% endif %}
-{% if not mon_containerized_deployment_with_kv %}
+{% if not mon_containerized_deployment_with_kv and not mon_containerized_deployment %}
 fsid = {{ fsid }}
 {% endif %}
 max open files = {{ max_open_files }}
 {% if common_single_host_mode is defined %}
 osd crush chooseleaf type = 0
 {% endif %}
-[client.libvirt]
-admin socket = {{ rbd_client_admin_socket_path }}/$cluster-$type.$id.$pid.$cctid.asok # must be writable by QEMU and allowed by SELinux or AppArmor
-log file = {{ rbd_client_log_file }} # must be writable by QEMU and allowed by SELinux or AppArmor
+{# NOTE (leseb): the blank lines in-between are needed otherwise we won't get any line break #}
+{% if groups[mon_group_name] is defined %}
+mon_initial_members = {% if groups[mon_group_name] is defined %}{% for host in groups[mon_group_name] %}{% if hostvars[host]['ansible_fqdn'] is defined and mon_use_fqdn %}{{ hostvars[host]['ansible_fqdn'] }}{% if not loop.last %},{% endif %}{% elif hostvars[host]['ansible_hostname'] is defined %}{{ hostvars[host]['ansible_hostname'] }}{% if not loop.last %},{% endif %}{% endif %}{% endfor %}{% endif %}
 
-[mon]
-{% if not mon_containerized_deployment_with_kv %}
-{% for host in groups[mon_group_name] %}
-{% if hostvars[host]['ansible_fqdn'] is defined and mon_use_fqdn %}
-[mon.{{ hostvars[host]['ansible_fqdn'] }}]
-host = {{ hostvars[host]['ansible_fqdn'] }}
-{% elif hostvars[host]['ansible_hostname'] is defined %}
-[mon.{{ hostvars[host]['ansible_hostname'] }}]
-host = {{ hostvars[host]['ansible_hostname'] }}
-{% endif %}
-# we need to check if monitor_interface is defined in the inventory per host or if it's set in a group_vars file
-{% if mon_containerized_deployment %}
-{% set interface = ["ansible_",ceph_mon_docker_interface]|join %}
-{% if interface in hostvars[host] and 'ipv4' in hostvars[host][interface] %}
-# user address from interface {{ ceph_mon_docker_interface }}
-mon addr = {{ hostvars[host][interface]['ipv4']['address'] }}
- {% elif hostvars[host]['monitor_address'] is defined %}
- # use host monitor address
-mon addr = {{ hostvars[host]['monitor_address'] }}
- {% elif monitor_address != "0.0.0.0" %}
- # use group_var monitor address
-mon addr = monitor_address
-{% endif %}
+mon_host = {% if groups[mon_group_name] is defined %}{% for host in groups[mon_group_name] %}{{ hostvars[host]['ansible_' + monitor_interface]['ipv4']['address'] }}{% if not loop.last %},{% endif %}{% endfor %}{% endif %}
```

Comment:
There are 2 different conditions, for containerized and not containerized deployment.